odl/trafos/fourier.py

```diff
@@ -1365,13 +1366,25 @@ def _call_pyfftw(self, x, out, **kwargs):
         return out
 
     @property
+    def adjoint(self):
+        """The adjoint Fourier transform."""
+        if self.domain.exponent != 2.0 or self.range.exponent != 2.0:
+            raise NotImplementedError(
+                'Adjoint only implemented for exponent=2')
```

Comment:
L^p is a Hilbert space only for p=2, so the notion of adjoint when 1 < p and p !=2 would have to be based on the Banach space adjoint. I have no idea what happens with these topological issues when considering a discretization, i.e. what happens with the Banach space adjoint in a discretization.